### DISCUSSION TITLE
auto-generated code causes build error

### DISCUSSION THREAD

**Discussion by JurjendeGroot** [MarimerLLC/csla#2234] [Questions]
I am using Csla v5.4.0 (in the ABP.IO framework)

When Building the solution in VS2019 (16.9.4) the following 'code' is generated in my Blazor project (in the obj folder) and causes a build-error (below). At this time since I'm still developing I just mark that line as comment and then all is well until I do another build. However I expect problems down the road when building in a devops pipeline...

Is there some way, an option or setting, to prevent this code being generated ? 

```
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

using System;
using System.Reflection;

[assembly: Microsoft.AspNetCore.Mvc.ApplicationParts.ApplicationPartAttribute("Csla.AspNetCore")]

// Generated by the MSBuild WriteCodeFragment class.`
```

**Build error:**
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0234	The type or namespace name 'Mvc' does not exist in the namespace 'Microsoft.AspNetCore' (are you missing an assembly reference?)	appFileManager.Blazor	C:\temp\....\Blazor.MvcApplicationPartsAssemblyInfo.cs	14	Active
```

**Comment by rockfordlhotka** [comment]
This code isn't generated by CSLA, it is being generated by ASP.NET (that is my guess anyway).

I suspect you are missing an assembly reference to `Microsoft.AspNetCore`.

**Reply by JurjendeGroot** [reply]
Hi Rocky,

I've added the nuget package Microsoft.AspNetCore but still get this error.

I already searched online for the class ApplicationPartAttribute and found:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.applicationparts.applicationpartattribute?view=aspnetcore-5.0

I added nuget package Microsoft.AspNetCore.App.Ref, this didn't solve the problem. I get this error after installing the package, so I removed it.
Severity	Code	Description	Project	File	Line	Suppression State
Error	NU1213	The package Microsoft.AspNetCore.App.Ref 5.0.0 has a package type DotnetPlatform that is incompatible with this project. 	Parrow.Blazor	E:\Gits.Git.Projects\Idas\PROTOTYPE-ABP43\src\Parrow.Blazor\Parrow.Blazor.csproj	1	

I added nuget package Microsoft.AspNetCore .Mvc and (later) Microsoft.AspNetCore.Mvc.Core but that made no difference. So I removed these packages.

Then I found this link:

https://github.com/dotnet/aspnetcore/issues/26724 and subsequently https://github.com/dotnet/aspnetcore/issues/24171 which suggest adding:

<GenerateMvcApplicationPartsAssemblyAttributes>false</GenerateMvcApplicationPartsAssemblyAttributes>

to the project properties. That seems to do the trick !

:-)
